File: 03.FrontendRTE/src/app/shared/architecture/data/validators/validatorutils.ts
import { isBefore } from '@shared/architecture/data/validators/asserts/isBefore';

export class ValidatorUtils {

  public static isDateInRange(date: Date | string, minusYears: number, plusYears: number): boolean {
    let begin = new Date();
    begin.setDate(Date.now());
    begin.setFullYear(begin.getFullYear() - minusYears);
    let end = new Date();
    end.setDate(Date.now());
    end.setFullYear(end.getFullYear() + plusYears);
    return isBefore(begin, date) && isBefore(date, end);
  }

  public static isDateAfterRWTHFounding(date: Date | string): boolean {
    let rwthFounding = new Date();
    rwthFounding.setFullYear(1870, 10, 10);
    return isBefore(rwthFounding, date);
  }

}

File: 04.BackendRTE/src/main/java/de/montigem/be/domain/rte/validator/ValidatorUtils.java
/* (c) https://github.com/MontiCore/monticore */

package de.montigem.be.domain.rte.validator;

import java.time.*;

public class ValidatorUtils {


  public static boolean isDateInRange(ZonedDateTime date, int minusYears, int plusYears) {
    return ZonedDateTime.now().minusYears(minusYears).isBefore(date) && date.isBefore(ZonedDateTime.now().plusYears(plusYears));
  }

  public static boolean isDateAfterRWTHFounding(ZonedDateTime date) {
    ZonedDateTime rwthFounding = ZonedDateTime.of(1870, 10, 10, 0, 0, 0, 0, ZoneId.of("GMT+2"));
    return rwthFounding.isBefore(date);
  }

}
